elements/lisk-p2p/src/p2p.ts

```diff
@@ -433,7 +446,14 @@ export class P2P extends EventEmitter {
 					this.emit(EVENT_NEW_PEER, incomingPeerInfo);
 				}
 
-				if (!this._newPeers.has(peerId) && !this._triedPeers.has(peerId)) {
+				if (
+					!this._newPeers.has(peerId) &&
+					!this._triedPeers.has(peerId) &&
+					!(
```

Comment:
Is this new condition needed?
If the node tries to connect to itself, it should be rejected on line `365` and it should never reach this code.

Reply:
That's true but somehow it's still being added to `newPeers` list reported by @shuse2 so at some point the nonce is different for the same IP, is it because we are having two nonces (network and config)?

Reply:
Removed this condition as discussed, as it will be covered by nonce condition already